### ISSUE TITLE
Komprese

### ISSUE THREAD

**Issue by ondrejhlavacek** [keboola/output-mapping#20]
U slicovaných souborů je docela dramatický zrychlení uploadu do S3 (~25 %), když se soubory komprimují (zařídí Storage API Client). Vyzkoušel bych to pro všechny možný scénáře a zapnul všude, kde to dává smysl.

https://github.com/keboola/storage-api-php-client/issues/233#issuecomment-439025372
https://github.com/keboola/storage-api-php-client/issues/233#issuecomment-439049316

**Comment by ondrejhlavacek** [comment]
- [ ] zkusit na output-mappingu testy
- [ ] zkusit docker-runner testy
- [ ] zkusit nějakej větší job na testingu

**Comment by ondrejhlavacek** [comment]
Zapnul jsem kompresi a udělalo to tohle - nezměnilo se jméno souboru (takže úplně nevím, co se děje) a nesedí velikost.

![image](https://user-images.githubusercontent.com/497675/49253901-0e43e580-f428-11e8-827a-f12fd82caf2f.png)

vs

![image](https://user-images.githubusercontent.com/497675/49253915-1865e400-f428-11e8-83e9-4d26fa179f95.png)


Slicovaný jsou o něco lepší, tam aspoň sedí velikost. 

![image](https://user-images.githubusercontent.com/497675/49253854-e5235500-f427-11e8-938d-9e8f9369186f.png)

vs 

![image](https://user-images.githubusercontent.com/497675/49253870-f5d3cb00-f427-11e8-9e5e-f008b12e9310.png)

**Comment by ondrejhlavacek** [comment]
Navrhnul bych pro přehlednost v Storage API klientovi přidat koncovku `.gz` souborům, co se komprimují v klientovi a u neslicovaných fixnout velikost.

**Comment by ondrejhlavacek** [comment]
Chyba byla tím, že jsem v docker runneru pustil pouze `composer update keboola/output-mapping`, kterej nenainstaloval poslední verzi storage api klienta. Divný je, že některý slicy seděly a jiný ne.

**Comment by ondrejhlavacek** [comment]
Neslicovaný soubory se defaultně komprimují, takže jsem mohl udělat jednoduchej end-end benchmark. Tři CSVčka z S3, stahovaný nekomprimovaně a pak poslaný do tabulky jako slicovaný tabulky (nekomprimovaně) nebo neslicovaný (komprimovaně). Výsledky jasný, nekomprimovaný job 11 (https://connection.keboola.com/admin/projects/5446/jobs/467560099?q=) minut, komprimovaný 6 minut (https://connection.keboola.com/admin/projects/5446/jobs/467560110?q=).

Teď to jen rozmotat a zpět smotat, protože uplně přesně nevím, kde se ta komprese u neslicovaných souborů bere, v output mappingu to není, takže to asi musí nativně dělat storage api klient.

**Comment by ondrejhlavacek** [comment]
No, jenže ten projekt má zapnutou tu deferred featuru, takže se všechny uploady prováděj v keboola/output-mapping a na klientovi jsou jen importní joby. 

https://github.com/keboola/output-mapping/blob/5d92a3a24a7ec1db36ac4834e1f43fda68b1c7a3/Writer/Writer.php#L614

A tady to je, takže vyřešeno, nenapadlo mě si double checknout, jestli to opravdu output-mapping nedělá. S3 extraktor, na kterým jsem to hlavně testoval, všechno (i 1 soubor) převádí na slicy, to mě taky trochu svedlo z cesty.

**Comment by odinuv** [comment]
jo, tak to je 1:1 kopie te metody akorat s pridanym parametrem, tak hlavne, ze se to vyjasnilo

**Comment by ondrejhlavacek** [comment]
https://keboola.slack.com/archives/C31U6BGJC/p1543532807021200

Neslicovaný soubory to i správně pojmenuje

https://github.com/keboola/storage-api-php-client/blob/6b01f17d008a1a6161306ba24609252024e928b8/src/Keboola/StorageApi/Client.php#L1273-L1278
